scripts/rqtl2_wrapper.R

```diff
@@ -0,0 +1,447 @@
+# This script file contains  the an implementation of qtl mapping using r-qtl2
+# For r-qtl1 implementation see: Scripts/rqtl_wrapper.R
+# load the library
+
+library(qtl2)
+library(rjson)
+library(stringi)
+library(stringr)
+library(optparse)
+
+
+
+option_list <- list(
+  make_option(c("-c", "--cores"), type="integer", default=1, help="No of cores to use while making
+  computation"),
+make_option(c("-i", "--input_file"), action="store", default=NULL, type='character', help="a yaml or json file with required data to create the cross file"),
+make_option(c("-p", "--nperm"), type="integer", default= 1,  action="store_true", help="No  of permutations "),
+ make_option(c("-d", "--directory"), action = "store", default = NULL, type = "character", help="Temporary working directory: should also host the input file ."),
+ make_option(c("-m", "--method"), action = "store", default = "HK", type = "character", help="Scan Mapping Method - HK (Haley Knott), LMM( Linear Mixed Model ), LOCO (Leave one Chromosome Out)"),
+make_option(c("-o", "--output_file"), action="store", default=NULL, type='character', help="a file name of where to write the output json results"),
+  make_option(c("--pstrata"), action="store_true", default=NULL, help="Use permutation strata")
+)
+
+
+opt <- parse_args(OptionParser(option_list=option_list))
+# check for mandatory args
+if(is.null(opt$directory)){
+stop("You need  to specify  a working temporary directory which should have be base_dir for the input file")
+}
```

Comment:
It may be better to group the default args checks together, to help with code reading. Can we move the checks for `opt$input_file` and `opt$output_file` here?

Reply:
Done